core/src/main/java/org/elasticsearch/search/suggest/completion/CompletionSuggestParser.java

```diff
@@ -158,6 +160,14 @@ public CompletionSuggestParser(CompletionSuggester completionSuggester) {
                 } else {
                     throw new IllegalArgumentException("suggester [completion] doesn't support field [" + fieldName + "]");
                 }
+            } else if (token == XContentParser.Token.START_ARRAY) {
+                if ("payload".equals(fieldName)) {
+                    while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
```

Comment:
Thanks for the pointer, I have made it stricter and also added tests to ensure it errors out on malformed input